lib/ast/definition.cpp

```diff
@@ -104,7 +104,12 @@ void kore_definition::insert_reserved_symbols() {
   add_module(std::move(mod));
 }
 
-SubsortMap kore_definition::get_subsorts() const {
+SubsortMap kore_definition::get_subsorts() {
```

Comment:
I don't follow the logic in this function; all it needs to do is compute the transitive closure _once_ and cache that result. Why are you recomputing it?

Reply:
The problem here is that we need the non-transitive subsorts set to compute the supersorts. However, on second thought, we could just calculate it in the same way and cache everything right!